[docs](docs)Add Docker Image build documentation

FreeOnePlus opened the pull request:
# Proposed changes

Add Docker Image build documentation

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

FreeOnePlus committed:
fix doc

FreeOnePlus committed:
Merge branch 'master' of https://github.com/apache/doris

FreeOnePlus committed:
Merge branch 'master' of https://github.com/apache/doris

FreeOnePlus committed:
Merge branch 'apache:master' into master

FreeOnePlus committed:
Merge branch 'apache:master' into master

FreeOnePlus committed:
Merge branch 'apache:master' into master

englefly committed:
[test](nereids) add filter estimation ut cases (#14293)

fix a bug for filter estimation, in pattern of A>10 and A<20.

FreeOnePlus committed:
add construct-docker-image.md

FreeOnePlus committed:
add construct-docker-image.md

hf200012 committed:
datax doris writer doc fix (#14344)

morrySnow committed:
[enhancement](explain) compress descriptor table explain string (#14152)

1. compress slot descriptor explain string to one row
2. remove unmaterialized tuple descriptor and slot descriptor

before this PR descriptor table explain string is like this:
```
TupleDescriptor{id=0, tbl=lineitem, byteSize=176, materialized=true}
  SlotDescriptor{id=0, col=l_shipdate, type=DATEV2}
    parent=0
    materialized=true
    byteSize=4
    byteOffset=0
    nullIndicatorByte=0
    nullIndicatorBit=-1
    nullable=false
    slotIdx=0

  SlotDescriptor{id=1, col=l_orderkey, type=BIGINT}
    parent=0
    materialized=true
    byteSize=8
    byteOffset=24
    nullIndicatorByte=0
    nullIndicatorBit=-1
    nullable=false
    slotIdx=6
```

after this PR descriptor table explain string is like this:
```
TupleDescriptor{id=2, tbl=lineitem}
  SlotDescriptor{id=1, col=l_extendedprice, type=DECIMAL(15,2), nullable=false}
  SlotDescriptor{id=2, col=l_discount, type=DECIMAL(15,2), nullable=false}
```

dataroaring committed:
[improvement](test) add conf for pipline (#14254)

Add conf used by pipline to git, then we will change conf of pipeline
via pr like code commit.

wsjz committed:
[feature-wip](multi-catalog) support iceberg v2(step 1) (#13867)

Support position delete(part of).

yiguolei committed:
[cherry-pick](scanner) using avg rowset to calculate batch size instead of using total_bytes since it costs a lot of cpu (#14345)

Co-authored-by: yiguolei <yiguolei@gmail.com>

jacktengg committed:
[fix](runtimefilter) fix heap-user-after-free of runtime filter merge (#14362)

Kikyou1997 committed:
[fix](statistics) statistics inaccurate after analyze same table more than once (#14279)

If a table already been analyzed, then we analyze it again, the new statistics would larger than expected since the incremental would contain the values from table level statistics since the SQL lack the predication for the nullability of part_id

AshinGau committed:
[test](multi-catalog)Regression test for external hive orc table (#13762)

Add regression test for external hive orc table. This PR has generated all basic types support by hive orc, and create a hive external table to touch them in docker environment.
Functions to be tested:
1. Ensure that all types are parsed correctly
2. Ensure that the null map of all types are parsed correctly
3. Ensure that the `SearchArgument` of `OrcReader` works well
4. Only select partition columns

Jibing-Li committed:
[feature-wip](multi-catalog) Collect external table statistics (#14160)

Collect HMS external table statistic information through external metadata.
Insert the result into __internal_schema.column_statistics using insert into SQL.

morningman committed:
[improvement](multi-catalog)(cache) invalidate catalog cache when refresh (#14342)

Invalidate catalog/db/table cache when doing
refresh catalog/db/table.

Tested table with 10000 partitions. The refresh operation will cost about 10-20 ms.

HappenLee committed:
[Neried](Profile) Add projection timer for neried (#14286)

Gabriel39 committed:
[feature](join) support nested loop semi/anti join (#14227)

spaces-X committed:
[fix](profile)(AggNode) fix the GetResultsTime is always zero (#14366)

add scoped_timer in _serialize_with_serialized_key_result

liaoxin01 committed:
[Enhancement](sequence-column) optimize the use of sequence column (#13872)

When you create the Uniq table, you can specify the mapping of sequence column to other columns.
You no longer need to specify mapping column when importing.

xinyiZzz committed:
[enhancement](memtracker)  Use proc/meminfo MemAvailable to control memory and optimize MemTracker log printing (#14335)

morrySnow committed:
[fix](Nereids) DateTimeType migrate to DateType is wrong when hour, minute and second all zero (#14327)

1. fix DateTimeType migrate to DateType is wrong when hour, minute and second all zero
2. add TPC-H regression test with DATEV2 type

adonis0147 committed:
[chore](macOS) Fix BE UT (#14307)

#13195 left some unresolved issues. One of them is that some BE unit tests fail.
This PR fixes this issue. Now, we can run the command ./run-be-ut.sh --run successfully on macOS.

keanji-x committed:
[feature](Nereids): Add subgraph enumerator #14291

Add subgraph enumerator to find the best plan

For DPHyp, we need an enumerator for all csg-cmp pairs to find the best plan

morrySnow committed:
[compatibility](Nereids) process escape in string literal (#14294)

starocean999 committed:
[enhancement](be)limit mem cost to 16m when pre serialize keys in agg node (#14321)

* [enhancement](be)limit mem cost to 16m when pre serialize keys in agg node

* use only one chunk memory when serializing keys in agg node

BePPPower committed:
[feature](table-valued-function) support `hdfs()` tvf (#14213)

This pr does two things:
1. support `hdfs()` table valued function.
2. add regression test

morningman committed:
[improvement](ctas) use string type for varchar/char/string (#14382)

When executing create table as select stmt,
the varchar/char/string type of column in created table will be unified to string type.

Because when select from external table (mysql/pg, etc), the length of varchar in external database
is calculated by "char" length, not "byte" length.
So if there is a column with varchar(10) in external table, then there will be a same varchar(10)
in created table. But the byte length of data in external table may be larger than 10, causing failure of CTAS.

Change to string will not impact performance of the capacity of disk storage.
And notice that if a string type column is the first column, it will be changed to varchar(65535),
because we do not allow string type column as sort key column.

BiteTheDDDDt committed:
[Bug](runtime filter) fix minmax filter not copy rightly on shared hash join (#14367)

fix minmax filter not copy rightly on shared hash join

hf200012 committed:
[fix](datax)doris writer write error (#14276)

* doris writer write error

carlvinhust2012 committed:
[optimization](array-type) optimize the export precision of floating point numbers (#14261)

Co-authored-by: hucheng01 <hucheng01@baidu.com>

nextdreamblue committed:
[fix](type) fix DECIMAL scale when cast function on fe (#12877)

before:
MySQL [test]> select cast('135.759999999' as DECIMAL(10,3));
+----------------------------------------+
| CAST('135.759999999' AS DECIMAL(10,3)) |
+----------------------------------------+
| 135.759999999 |
+----------------------------------------+
1 row in set (0.00 sec)

now:
MySQL [stage]> select cast('135.759999999' as DECIMAL(10,3));
+----------------------------------------+
| CAST('135.759999999' AS DECIMAL(10,3)) |
+----------------------------------------+
| 135.759 |
+----------------------------------------+
1 row in set (0.01 sec)

xinyiZzz committed:
[enhancement](memory) Support Jemalloc metrics and default allocator changed to Jemalloc  (#14384)

liaoxin01 committed:
[fix](broker-load) fix that broker load don not set be exec version and limit node channel memory (#14399)

englefly committed:
[refactor](nereids) remove ColumnStatistics.UNKNOWN from StatsDerive (#14343)

ColumnStatistics.UNKNOWN can be replaced by ColumnStatistics.DEFAULT

924060929 committed:
[Enhancement](Nereids) Refactor AggregateFunction and support explain plan (#14380)

# Proposed changes

- Refactor AggregateFunction
    1. AggregateFunction implement ComputeSignature
    3. Add a CustomSignature to dynamic compute signature, we can check input type and compute implicit cast type in the `customSignature` method
    2. Add PartialAggType to record some type information before disassemble aggregate
    4. Refine and create a custom catalog function when translate AggregateFunction, without `finalizeForNereids`
-  Support explain plan
    1. explain parsed plan select ...
    5. explain analyzed plan select ...
    6. explain rewritten/logical plan select ...
    7. explain optimized/physical plan select ...
    8. explain all plan select ...

adonis0147 committed:
[chore](third-party) Use GNU official mirror to boost the download speed (#14358)

According to the description in https://www.gnu.org/server/mirror.html, using the address http://ftpmirror.gnu.org/ to download GNU packages is recommended. It can boost the download speed worldwide.

lsy3993 committed:
[test](jdbc external table) add new jdbc mysql external table (#14323)

eldenmoon committed:
[feature-array](array-function) add array constructor function `array()` (#14250)

* [feature-array](array-function) add array constructor function `array()`

```
mysql>  select array(qid, creationDate) from nested_c_2  limit 10;
+------------------------------+
| array(`qid`, `creationDate`) |
+------------------------------+
| [1000038, 20090616074056]    |
| [1000069, 20090616075005]    |
| [1000130, 20090616080918]    |
| [1000145, 20090616081545]    |
+------------------------------+
10 rows in set (0.01 sec)
```

morningman committed:
[fix](parquet-reader) fix stack-use-after-return error (#14411)

gnehil committed:
[improvement](common) table name length exceeds limit error message (#14368)

For the table name check, the regular match error and the length exceeds the limit, both of which display the message "Incorrect table name 'xxx'. Table name regex is 'xxx'".
Obviously, the message cannot clearly point out what kind of error it is.
So it is a better way to separate the two error messages.

FreeOnePlus committed:
[refactor](planner) remove the limit return rows of order by (#12478)

Originally, Order By Limit returned a maximum of 65535 rows of data by default during the query,
but now many businesses do not apply this limit.
It is necessary to add larger data after the query statement to complete the full data query,
which is extremely inconvenient, so adjustments have been made.

At the same time, I added the variable DEFAULT_ORDER_BY_LIMIT to the SessionVariable,
the default value is -1, if the user does not use the LIMIT keyword or the LIMIT value is a negative integer,
the default query return value is Long.MAX_VALUE. If the corresponding maximum query value is set,
the number of data items is returned according to the maximum query value or the value followed by the
LIMIT keyword.

morningman commented on docs/zh-CN/docs/install/construct-docker-image.md:
````diff
@@ -0,0 +1,258 @@
+   ```shell
+   # 下载 FE 二进制包
+   cd ./docker-build/fe/resource
+   wget https://mirrors.tuna.tsinghua.edu.cn/apache/doris/x.x/x.x.x-rc0x/apache-doris-fe-x.x.x-bin.tar.gz
````
这里直接引用官网下载页面即可

morningman commented on docs/zh-CN/docs/install/construct-docker-image.md:
````diff
@@ -0,0 +1,258 @@
+
+```shell
+# 创建指定网桥
+docker network create --subnet=172.20.80.0/24 dorisNetwork
````
这里增加下ip的说明？比如是否都可以使用 172.20.80.0/24 这个配置？

morningman commented on docs/zh-CN/docs/install/construct-docker-image.md:
````diff
@@ -0,0 +1,258 @@
+```
+
+可以通过以下命令来创建一个 FE 容器和一个 BE 容器来测试构建是否成功
+
````
FE 和 BE 的数据目录，conf目录，是否可以挂在到宿主机？
可以的话，给出说明。

caiconghui committed:
[enhancement](rewrite) Remove unused wide common factors to improve scan performance in ExtractCommonFactorsRule (#14381)

* [enhancemeng](sql) Remove unused wide common factors to improve scan performance in ExtractCommonFactorsRule

* fix regression test

Co-authored-by: caiconghui1 <caiconghui1@jd.com>

hello-stephen committed:
[tools](tpch)fix invalid download url (#14329)

Gabriel39 committed:
[refactor](decimalv3) Refine code for DecimalV3 (#14394)

lsy3993 committed:
[test](hive catalog)add hive catalog test case (#14217)

zhannngchen committed:
[fix](exec) fix thread token shutdown (#14418)

Fix Thread pool token was shut down error.
This is because when there are more than 1 fragment of a query on one BE, the thread token maybe
reset incorrectly, causing thread token shutdown earlier.
cherry-pick from master
Introduced from #13021

dataroaring committed:
[fix](test) sync conf used in pipeline and in repository (#14414)

morningman committed:
(improvement)[doc] add document version tag instruction (#14406)

hf200012 committed:
[typo](docs)Performance test documentation update (#14147)

* Performance test documentation update

HappenLee committed:
[Docs](function) Add some function do not in sidebars (#14426)

hf200012 committed:
[typo](docs)benchmark doc fix number (#14427)

zhangstar333 committed:
[bug](test) fix regression test of jdbc postgresql table core (#14417)

BiteTheDDDDt committed:
[Optimize](predicate)  update inplace on VcompoundPred  (#14402)

select count(*) from lineorder where lo_orderkey<100000000 OR lo_orderkey>100000000 AND lo_orderkey<200000000 OR lo_orderkey >200000000;

0.6s -> 0.5s

zhouaoe committed:
[Enhancement] Doris broker support aliyun-oss #13665 (#14305)

englefly committed:
[feature](tools)profile viewer (#14429)

It is a painful work to read profile, especially there are multi-parallel instances.
This tool helps us to grasp the main information of profile in a graphical view.

The profile is represented by a tree.
Sql operation nodes contains operation type(join, scan...), its node id, its fragment id. The number on the arrow edge means how many rows output by child node. This tool will sum the output rows of the same node in multi-parallel instances, that is if there are 4 parallel instance, and each ScanNode on lineitem table output 10 rows, the label on the arrow beginning with ScanNode(lineitem) is 40.

Here is a demo for tpch Q2
tpch q2 profile viewer

Issue Number: close #xxx

nextdreamblue committed:
[fix](insert) fix insert into table which contains column name prefix mv_ (#14361)

zhannngchen committed:
[improvement](load) add a log when close OlapTableSink with error (#14257)

zy-kkk committed:
[Feature](JDBC)support clickhouse jdbc external table (#14244)

adonis0147 committed:
[chore](macOS) Fix linker errors (#14410)

FreeOnePlus committed:
change docs
